graphics/surface.cpp

```diff
@@ -19,6 +19,11 @@
  *
  */
 
+#ifdef ATARI
```

Comment:
All this commit is really invasive.
I am not sure we want to add platform specific code here.

Reply:
I'd say you don't. Ideally, if every copy function would call just `Graphics::`, that would make it much cleaner. For instance I wouldn't see a problem with `Graphics::keyBlit` having `#ifdef`ed for hardware acceleration in `blit.cpp`.  So maybe the copy functions could be moved into the `Graphics` NS?

Reply:
Ideally, any platform specific code should be implemented as part of `OSystem`, since having substancial platform-specific ifdefs doesn't scale very well.

For now, what might be best is to leave these changes out for now, and then once this PR is merged we can discuss how best to approach this in a follow-up PR.

Reply:
Indeed, these changes are way too invasibe. If you truly want overriding the `Graphics::Surface` behaviour, I'd suggest then putting all the differences into a separate file and including it only on your platform. And yes, you will then be doomed to sync the implementations every time we change surface.cpp

Reply:
@sev- wouldn't then be easier to follow up on the idea of creating a simple `blit()` function in `Graphics` NS so I could put only one or two `#ifdefs` in `blit.cpp` and implement my blit routines within `backends/graphics/atari/atari-blit.cpp` ?

Reply:
Please explain your idea a bit. We already have `graphics/blit.cpp`. If you mean moving out the blitting code from the Surface class, yes, it is possible. Just name the file like `surface-blit.cpp`, so you could substitute on your platform, e.g. have next to it a file `surface-blit-atari.cpp` with the replaced methods and then use `ifdef ATARI` in `module.mk`

Reply:
You know what, give me a few days -- I have something in the works which a) is closer to what you have proposed earlier (separate surface.cpp file, in my case just a couple of functions, esp. if https://github.com/scummvm/scummvm/pull/4727 gets merged) and b) works better for me anyway - I can't take the `Graphics` path after all as I need to override also `create` and `free` so it's not worth it.

However what would be worth having is a more flexible `Surface` class. Right now only `ManagedSurface` allows you to override various functions but on the other hand it has nearly all of its variables private, so it's not that useful for my purposes. Plus it is focused more on the engine usage (see its update functions), not in backends. So ideally, if I could override create, free, copyRectToSurface... that would make it really flexible for many tasks. But that's a discussion for another time I guess...

Reply:
@sev- what do you think about https://github.com/mikrosk/scummvm/commit/fad16e6bf5a22ea9f307f032d9bd1813541521a9 ? That makes one `#ifdef` in surface.cpp and one `#ifdef` in blit.cpp. I have a more advanced version in the works but I think it's better to finalise this PR process first before updating X files again. So this just moves functionality around, assuming that #4727 gets merged.

Reply:
Yes, I like that approach. Only one small thing: please add a comment above #iffef, briefly explaining that Atari has customer definitions in the file X

Reply:
Done.